addons/product_images/wizard/product_fetch_image_wizard.py

```diff
@@ -0,0 +1,264 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import re
+from datetime import timedelta
+
+import requests
+from requests.exceptions import ConnectTimeout, HTTPError, ReadTimeout
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class ProductFetchImageWizard(models.TransientModel):
+    _name = 'product.fetch.image.wizard'
+    _description = "Fetch product images from Google Images based on the product's barcode number."
+
+    @api.model
+    def default_get(self, fields_list):
+        # Check if the cron is already triggered and raise an UserError if so
+        cron_id = self.env.ref('product_images.ir_cron_fetch_image').id
+        date = self.env['ir.cron.trigger'].search(
+            [('cron_id', '=', cron_id)], limit=1, order="call_at desc"
+        ).call_at
+        if date:
```

Comment:
@Julien00859 is reading `call_at` even useful? Are there situations where crons have triggers without dates *and they should be ignored*?

Otherwise this can just be a `search_count`, which would probably be cheaper.

Reply:
I should have a deeper look because this code seems very odd, not mentioning ir.cron.trigger is admin-only

Reply:
`call_at` was previously used in a `UserError` just below. Since we decided to not show the date anymore, it is indeed useless and should be removed. I'll adapt the code with `search_count`.

This functionality is intended to be used only by admin so it shouldn't be a problem that `ir.cron.trigger` is admin-only.

Reply:
Feedback from a discussion we had at work,

TL;DR: their approach is the right one, they need to ensure there is at most 1 trigger for their cron.

The google API implements two ratelimits: they allow to fetch up to `x` images per minute per API key and `y` per day. You can pay to increase x and y. Because of this when you want to fetch, say, 1000 images you need a precise control on when to query google.

1) When there is no image to fetch, you should not query google (cron should be disabled or should not do anything)
2) When there are images to fetch, you should query google for `x` images then wait 1 minute before fetching `x` more.
3) When there are images to fetch but you you downloaded `y` already today, you have to wait 1 day before fetching more.

Their solution to this problem is to use a trigger-only cron with the additionnal constraint that there can only be one trigger at a time. The rationnal is as follow:

When the user asks for images, the initial trigger is created `at=now`, the cron is executed asap and fetches `x` records, it then re-trigger itself `at=now + 1m`. At some point google will respond with a http error 4xx meaning the user reached its daily limit, at that moment the cron re-trigger itself `at=now + 1d`.

Because the rate-limit is controlled thanks to the `at` mechanism of triggers **it is very important that there is at most 1 trigger at a time** (technically two: the one that currently executing the cron and one set in the future). IMO this constraint should be enforced thanks to a python contraint.

Reply:
@Julien00859  I've added a python constraint according to your recommendations :)